src/main.rs

```diff
@@ -30,9 +35,21 @@ fn word_wrap(paragraph: &str, line_length: usize) -> Vec<String> {
 	result
 }
 
-fn main() {
+fn main() -> Result<()> {
+	color_eyre::install()?;
+
 	let args = Cli::parse();
-	let mut lines = word_wrap(&args.message, 45);
+	let (cols, _) = terminal::size()?;
+
+	let width = match args.width {
+		Some(w) => match u16::from(w) >= cols - 5 {
+			true => cols - 5,
+			false => w.into(),
+		},
+		None => 45,
+	};
+
+	let mut lines = word_wrap(&args.message, width.into());
```

Comment:
```suggestion
	let mut lines = word_wrap(&args.message, width as usize);
```

This will make the types more apparent